sentinelhub/areas.py

```diff
@@ -230,39 +193,51 @@ class BBoxSplitter(AreaSplitter):
     area. If specified by user it can also reduce the sizes of the remaining bounding boxes to best fit the area.
     """
 
-    def __init__(self, shape_list, crs, split_shape, **kwargs):
+    def __init__(
+        self,
+        shape_list: Iterable[Union[Polygon, MultiPolygon, _BaseGeometry]],
+        crs: CRS,
+        split_shape: Union[int, Tuple[int, int]],
+        **kwargs: Any,
+    ):
         """
         :param shape_list: A list of geometrical shapes describing the area of interest
-        :type shape_list: list(shapely.geometry.multipolygon.MultiPolygon or shapely.geometry.polygon.Polygon)
         :param crs: Coordinate reference system of the shapes in `shape_list`
-        :type crs: CRS
         :param split_shape: Parameter that describes the shape in which the area bounding box will be split.
             It can be a tuple of the form `(n, m)` which means the area bounding box will be split into `n` columns
             and `m` rows. It can also be a single integer `n` which is the same as `(n, n)`.
-        :type split_shape: int or (int, int)
         :param reduce_bbox_sizes: If `True` it will reduce the sizes of bounding boxes so that they will tightly fit
             the given area geometry from `shape_list`.
-        :type reduce_bbox_sizes: bool
         """
+        self.split_shape = self._parse_split_parameters(split_shape)
         super().__init__(shape_list, crs, **kwargs)
 
-        self.split_shape = self._parse_split_parameters(split_shape)
+    @staticmethod
+    def _parse_split_parameters(split_parameter: Union[int, Tuple[int, int]]) -> Tuple[int, int]:
```

Comment:
Hm, it would be nice to unify the 3 almost the same methods but its not a high priority. I'm wondering if you can do that with TypeVars, something like:

```Python
def _parse_split_parameters(parameter: Union[_Type, Tuple[_Type, _Type]]) -> Tuple[_Type, _Type]:
```

Although this type is then also used for checking in the method, so the method might need a parameter to specify which type you want. Then the new type parameter would have to be typed with the same type var and I'm not sure if that is possible. :sweat_smile: 

So if you don't find a nice way you can just leave it.

Reply:
I can try to conjure up something with `override`. But I would probably need `Literal`, not entirely sure. TypeVars are probably not strong enough here as you noted yourself.

Reply:
I am actually going to try and do this, it seems like `@overload` is pretty strong

Reply:
Sadly, since we're not 3.8 and there is no neat supertype for float+int, I had to make some sacrifices.
Since the parameters are already typed, the functions type can be fairly non-complex, just needs a bit of generics.

Some confusion also follows from the fact that in [PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower) it is stated that `float` is the proper annotation for `float | int` types, but `isinstance(1, float)` fails. Therefore the type validation was decoupled from the type signature of the function. Once we go to 3.8 we can adjust this via overloading, until then it is what it is.

Reply:
Ok, looks good :+1: